R/files.R

```diff
@@ -0,0 +1,117 @@
+#' Checks if the value is a file
+#'
+#' @param value the value to check
+#' @param allow_null if TRUE, allow the value NULL. If FALSE (default) do not
+#'                   allow NULL.
+#' @examples
+#' \dontrun{
+#' # For assertion
+#' assertthat::assert_that(qscheck::is_file(f))
+#' # For check
+#' if (qscheck::is_file(f)) {}
+#' }
+#'
+#'
+#' @concept file
+#' @export
+is_file <- function(
+    value, allow_null = FALSE) {
+
+  res <- inspect_is_file(value, allow_null)
+  return(res$valid)
+}
+
+assertthat::on_failure(is_file) <- function(call, env) {
+  value <- callget(call, env, "value", NULL)
+  allow_null <- callget(call, env, "allow_null", FALSE)
+  res <- inspect_is_file(value, allow_null)
+
+  msg <- paste0(
+    deparse(call$value),
+    snippet_must_be("file"),
+    snippet_null(allow_null),
+    ". ", res$reason
+  )
+
+  return(msg)
+}
+
+inspect_is_file <- function(value, allow_null) {
+  if (is.null(value)) {
+    if (allow_null) {
+      return(success())
+    } else {
+      return(failure("Passed value cannot be NULL"))
+    }
+  }
+
+  if (!(file.exists(value))) {
+    return(failure(
+      paste(
+        "Passed value is not a file"
+      )
+    ))
+  }
+
+  return(success())
+
+}
+
+#' Checks if the value is a directory
+#'
+#' @param value the value to check
+#' @param allow_null if TRUE, allow the value NULL. If FALSE (default) do not
+#'                   allow NULL.
+#' @examples
+#' \dontrun{
+#' # For assertion
+#' assertthat::assert_that(qscheck::is_path(f))
+#' # For check
+#' if (qscheck::is_path(f)) {}
+#' }
+#'
+#'
+#' @concept directory
+#' @export
+is_path <- function(
+    value, allow_null = FALSE) {
+
+  res <- inspect_is_path(value, allow_null)
+  return(res$valid)
+}
+
+assertthat::on_failure(is_path) <- function(call, env) {
+  value <- callget(call, env, "value", NULL)
+  allow_null <- callget(call, env, "allow_null", FALSE)
+  res <- inspect_is_path(value, allow_null)
+
+  msg <- paste0(
+    deparse(call$value),
+    snippet_must_be("valid directory"),
+    snippet_null(allow_null),
+    ". ", res$reason
+  )
+
+  return(msg)
+}
+
+inspect_is_path <- function(value, allow_null) {
+  if (is.null(value)) {
+    if (allow_null) {
+      return(success())
+    } else {
+      return(failure("Passed value cannot be NULL"))
+    }
+  }
+
+  if (!(dir.exists(value))) {
```

Comment:
a path is not a dir. A path is a string that complies with the rules of the path on the platform you are running on. A path may not even refer to an existent file or directory.